WordPressKit/AccountServiceRemoteREST.m

```diff
@@ -127,31 +127,54 @@ - (void)isPasswordlessAccount:(NSString *)identifier success:(void (^)(BOOL pass
 
 - (void)isEmailAvailable:(NSString *)email success:(void (^)(BOOL available))success failure:(void (^)(NSError *error))failure
 {
+    static NSString * const errorEmailAddressInvalid = @"invalid";
+    static NSString * const errorEmailAddressTaken = @"taken";
+    
     [self.wordPressComRestApi GET:@"is-available/email"
-       parameters:@{ @"q": email, @"format": @"json"}
-          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
-              if (!success) {
-                  return;
-              }
+                       parameters:@{ @"q": email, @"format": @"json"}
+                          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+        if (!success) {
+            return;
+        }
 
-              // If the email address is not available (has already been used)
-              // the endpoint will reply with a 200 status code and an JSON
-              // object describing an error.
-              // The error is that the queried email address is not available,
-              // which is our failure case. Test the error response for the
-              // "taken" reason to confirm the email address exists.
-              BOOL available = NO;
-              if ([responseObject isKindOfClass:[NSDictionary class]]) {
-                  NSDictionary *dict = (NSDictionary *)responseObject;
-                  available = [[dict numberForKey:@"available"] boolValue];
-              }
-              success(available);
+        if ([responseObject isKindOfClass:[NSDictionary class]]) {
+            NSString *error = [responseObject objectForKey:@"error"];
+            NSString *message = [responseObject objectForKey:@"message"];
+            
+            if ([error isEqualToString:errorEmailAddressInvalid]) {
+                NSError* error = [[NSError alloc] initWithDomain:AccountServiceRemoteErrorDomain
+                                                            code:AccountServiceRemoteEmailAddressInvalid
+                                                        userInfo:@{
+                                                            @"response": responseObject,
+                                                            NSLocalizedDescriptionKey: message,
+                                                        }];
+                failure(error);
```

Comment:
Same comment as below. I've noticed there's no checks, here, for the scenarios in which `failure = nil` (but we do have one in the failure callback).

Just noting this, in case you wanna change it.

Reply:
Addressed.